### PR TITLE
test: Migrate some json tests to snapbox

### PR DESCRIPTION
**Pull request by epage**
### What does this PR try to resolve?

This is part of #14039

### Additional information

This was unblocked because of assert-rs/snapbox#350

### CODE REVIEW

File: tests/testsuite/doc.rs [hunk 1]
```diff
@@ -1879,24 +1879,25 @@ fn doc_message_format() {
 
     p.cargo("doc --message-format=json")
         .with_status(101)
-        .with_json_contains_unordered(
```
Comment (weihanglo) [inline tests/testsuite/doc.rs:1882]: Does the snapbox patch now correctly handle unordered JSON lines, or this test doesn't need unordered assertion?

Comment (epage) [reply tests/testsuite/doc.rs:1882]: If by "now correctly" you mean "implicitly", then no, as that would not be correct.  I am not seeing anything in this test case that needs `.unordered()` applied.

Comment (weihanglo) [reply tests/testsuite/doc.rs:1882]: I meant to comment under `doc_json_artifacts` but that one doesn't seem to need unordered either.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
@bors r+

Thanks!

**Pull request merged**